### PR TITLE
 #163553291 create API endpoint for user signup

### PR DESCRIPTION
**Pull request by Jesse-efe**
#### What does this PR do?
create API endpoint for user signup
#### Description of Task to be completed?
user can create an account

user gets back a token on signup

#### How should this be manually tested?
- Pull this repository and cd into it
- follow the steps under installation in project readme
- Use Postman to test the routes api/v1/auth/signup
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
#163553291 
#### Screenshots
N/A
#### Questions:
N/A

### CODE REVIEW

File: source/controller/users.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      text: 'SELECT * FROM users WHERE email = $1',
+      values: [email],
+    };
+
+    try {
+      const result = await pool.query(query);
+      if (result.rowCount !== 0) {
+        return res.status(400).json({
+          status: 400,
+          error: 'You are already a rigistered user please signin',
```
Comment (akinwale-habibullah) [inline source/controller/users.js]: Why not: A user with this email already exists.

Comment (Jesse-efe) [reply source/controller/users.js]: OK I would change it

**Review by akinwale-habibullah — COMMENTED** [review]
image

Your test for this feature is failing. Please look into it.


image


This is the signup endpoint's response. Here's another one

image

**Comment by akinwale-habibullah** [comment]
@Jesse-efe Please look into your validation.

**Comment by Jesse-efe** [comment]
Ok I will

**Pull request merged**